### PR TITLE
automate flux2 version upgrade

### PR DESCRIPTION
**Pull request by josecordaz**
**What changed?**
PR automation to upgrade flux

We would need to execute `.github/scripts/current_and_old_flux_versions.s` in case we want to debug the parameters passed in to the Github action creating the PR.

**Why?**
It would be easier to update flux

**How did you test it?**
I used the same yaml config on isolated private repo

**Release notes**

**Documentation Changes**

Example from test repo of how the PR will look like:
image
image

### CODE REVIEW

File: .github/workflows/upgrade-flux.yaml
Comment (jpellizzari) [inline .github/workflows/upgrade-flux.yaml:66]: I think we need to lock these to specific hashes now? @morancj can you confirm?

File: .github/workflows/upgrade-flux.yaml
Comment (jpellizzari) [inline .github/workflows/upgrade-flux.yaml:19]: Nit: we can lock this to prevent inconsistencies from happening in CI.

File: .github/workflows/upgrade-flux.yaml
Comment (jpellizzari) [inline .github/workflows/upgrade-flux.yaml:8]: Does this automatically run when someone pushes to `main`? I think we only want to run this on the `cron` job, once a day (if I am reading that correctly).

**Review by jpellizzari — COMMENTED** [review]
The job itself looks great! Just a question about the triggers

File: .github/workflows/upgrade-flux.yaml
Comment (josecordaz) [reply .github/workflows/upgrade-flux.yaml:8]: you are right

**Review by jpellizzari — COMMENTED** [review]
Sorry @josecordaz I should have caught this before, would you mind putting this logic into a script file that we can call from our laptops? The problem with putting this in a Github action is that we cannot run it locally.

In general, we want to be able to reproduce everything that CI does locally.

**Review by jpellizzari — APPROVED** [review]
LGTM, modulo one change: move the script to `tools/` and rename to `get-flux-update-pr.sh`

Also, I'm not super great at bash, so maybe we can get another set of :eyes: on this before merging.

File: .github/workflows/upgrade-flux.yaml
Comment (morancj) [reply .github/workflows/upgrade-flux.yaml:30]: Yes, please. [zgosalvez/github-actions-ensure-sha-pinned-actions](https://github.com/zgosalvez/github-actions-ensure-sha-pinned-actions) can help. For some commentary, see https://github.com/weaveworks/weave-gitops/pull/386.

File: tools/get-flux-update-pr.sh
Comment (jrryjcksn) [inline tools/get-flux-update-pr.sh:13]: I think this has the same issue in that it would fail if we reordered the file. Maybe something like:
`newContent=$(cat /tmp/dependencies.toml | tr '\n' '\r' | sed -e "/\[flux\]\rversion\=\"[0-9\.]*\"/{s//\[flux\]\rversion=\"$latestFluxVersion\"/;:a" -e '$!N;$!ba' -e '}' | tr '\r' '\n')`

**Review by jrryjcksn — APPROVED** [review]
A couple of comments but LGTM

File: tools/get-flux-update-pr.sh
Comment (morancj) [inline tools/get-flux-update-pr.sh:3]: Nice! Does `stoml` work well?

Comment (jrryjcksn) [reply tools/get-flux-update-pr.sh:3]: We've been using it for a long time, starting a while back in WKP. It's so simple that it's hard to know what could go wrong :-)

File: tools/get-flux-update-pr.sh
Comment (josecordaz) [reply tools/get-flux-update-pr.sh:13]: just changed `/tmp` with `/tools` and worked perfect, thanks 🙂

File: tools/get-flux-update-pr.sh
Comment (josecordaz) [reply tools/get-flux-update-pr.sh:3]: yes it does :)

**Pull request merged**
